lib/providers/password.js

```diff
@@ -16,31 +16,46 @@ var errors = require('../errors');
 function PasswordProvider (client, options) {
     // Jar for storing the session cookie
     this.jar = request.jar();
+    // storage of the CSRF Token as given in the response from the server
+    this.csrfToken = null;
+    this.csrfTokenLocation = 'x-csrf-token';
     this.auth = options;
     this.client = client;
 }
 
 /**
  * Attempts to authenticate with the given details. Resolves
- * with user information iff correct.
+ * with user information if correct.
  * @return {Promise}
  */
 PasswordProvider.prototype.attempt = function () {
     return this.client.request('post', '/users/login', { form: this.auth }).then(function (res) {
         if (res.statusCode !== 200) {
             throw new errors.AuthenticationFailedError(res);
-        } else {
-            return res;
         }
-    });
+        this.setCSRFToken(res.headers[this.csrfTokenLocation]);
```

Comment:
This isn't necessary. `this.client.request` delegates to `.getRequest` automatically